cbioportal_etl/scripts/diff_studies.py

```diff
@@ -105,85 +112,90 @@ def output_clinical_changes(
-                    file=out,
-                )
-                if attr not in update_delta_attr_cts:
-                    update_delta_attr_cts[attr] = 0
-                update_delta_attr_cts[attr] += 1
-                update_delta_ids.add(clinical_id)
-    return update_delta_ids, update_delta_attr_cts
-
-
-def table_to_dict(in_file: str, key: str, aggr_list: list) -> tuple[dict[str, dict], set, str, list[str], list[str] | None]:
+def split_data_file(
+        data_lines: list[str],
+        column_names: list[str],
+        key: str,
+        aggr_list: list,
+        current_data: dict[str,str],
+        shared_attrs: set[str],
+) -> tuple[dict[str, dict], list[str], dict[str, dict], list[str] | None]:
     """Take a text file and convert to dict with certain row value as primary, all other row values as subkeys
 
     Args:
-        in_file: str of input flat file name to parse
-        key: str of column name containing primary key to use to aggregate data
-        aggr_list: list of column names in which data values are are to be split and sorted to avoid a;b != b;a scenarios
+        data_lines: list of strings representing the lines of data
+        column_names: list of strings representing the column names
+        key: str of column name to use as primary key in dict for all other columns
+        aggr_list: list of column names in which data values are to be split and sorted to avoid a;b != b;a scenarios
+        current_data: dict containing the current dataset
+        shared_attrs: set containing the current attributes
     Returns:
-        data_dict: dict with flat file data converted as key value as primary key, associated column name as sub key, and row value as value
-        attributes: set of col names sans primary key
-        "\t".join(header) +"\n": str col names
-        data_clinical: list in_file read in as array of strings
-        big_head: str of header lines from in_file
-
+        delta_data
+        delta_lines
+        new_data
+        new_lines
     """
-    big_head, header, data_clinical = parse_file(
-        file_path=in_file, header_symbol="#", column_names=True,
-    )
-    data_dict: dict[str, dict] = {}
-    for entry in data_clinical:
+    delta_data: dict[str, dict] = {}
+    delta_lines: list[str] = []
+    new_data: dict[str, dict] = {}
+    new_lines: list[str] = []
+    for entry in data_lines:
         data: list = entry.rstrip("\n").split("\t")
         # Create a dictionary by zipping together the line with the header; sub in "NA" for empty columns
-        entry_dict: dict[str, str] = {k: ("NA" if v == "" else v) for k, v in zip(header, data)}
+        entry_dict: dict[str, str] = {k: ("NA" if v == "" else v) for k, v in zip(column_names, data)}
         # Aggregate attributes have multiple entries in a random order separated by a semicolon
         # Therefore, sort them so that when compared, no errors are triggered
         for aggr in aggr_list:
             if aggr in header:
```

Comment:
```suggestion
            if aggr in column_names:
```
IDE reports this is not defined, probably because it got renamed in the zip function from before